app/workers/metrics/record_db_and_es_record_counts_worker.rb

```diff
@@ -0,0 +1,64 @@
+# This worker checks the number of records for each Elasticsearch index and
+# compares it to the number of records in our database.
+#
+# For indexes that don't match a single model (no model or multiple models) we
+# need to implement a custom db_count method on the Search class to do the
+# counts for us
+module Metrics
+  class RecordDbAndEsRecordCountsWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :low_priority, retry: 5
+
+    # Once the Elasticsearch migration is complete we can update this value to
+    # Search::Cluster::SEARCH_CLASSES which will include all indexes.
+    SEARCH_CLASSES = [
+      Search::ChatChannelMembership,
+      Search::ClassifiedListing,
+      Search::Tag,
+    ].freeze
+
+    # Adjustable margin of error - this is how far off the index count can be
+    # from the database count before we raise an error
+    def perform(margin_of_error = 0)
+      SEARCH_CLASSES.each do |search_class|
+        margin_of_error = margin_of_error.to_i
```

Comment:
The JSON data types survive the serialization roundtrip, so you may not need the `to_i`.

> The Sidekiq server pulls that JSON data from Redis and uses JSON.load to convert the data back into Ruby types to pass to your perform method.

That said, should we maybe make this a percentage instead of an absolute number? Models we have more documents of, like articles, probably also have a higher turnover, so maybe also a higher chance for temporary drift? Not sure that's really the best approach, just throwing it out there.

Reply:
It's true that index workers have high priority but there could be a case where the queue is a bit overwhelmed. Maybe the default shouldn't be 0 or as @citizen428 suggsted it should be about percentages instead of relative numbers. Not sure. Maybe we can go ahead with this and if you start getting too many false positives notifications, you can tune this